pkg/policy/mapstate.go

```diff
@@ -1069,9 +1041,10 @@ func (ms *mapState) addKeyWithChanges(key Key, entry MapStateEntry, identities I
 		// Compare for datapath equalness before merging, as the old entry is updated in
 		// place!
 		datapathEqual = oldEntry.DatapathEqual(&entry)
+
 		oldEntry.Merge(&entry)
 		ms.insert(key, oldEntry, identities)
-	} else {
+	} else if !exists || entry.IsDeny {
```

Comment:
Can you comment why `entry.IsDeny` is special here?

Reply:
Added this comment:
```
		// Insert a new entry if one did not exist or a deny entry is overwriting an allow
		// entry.
```